cosmos_transfer1/diffusion/datasets/example_transfer_dataset.py

```
@@ -226,15 +269,210 @@ def __str__(self):
+                            video_name_emb = video_name[:-2]
+                        else:
+                            video_name_emb = video_name
+                        t5_embedding_path = os.path.join(self.t5_dir, f"{video_name_emb}.pkl")
+                        with open(t5_embedding_path, "rb") as f:
+                            t5_embedding = pickle.load(f)["pickle"]["ground_truth"]["embeddings"]["t5_xxl"]
+                    else:
+                        # use camera prompt
+                        t5_embedding = self.prefix_t5_embeddings[view_key]
+                    t5_embedding = torch.from_numpy(t5_embedding)
+                    if t5_embedding.shape[0] < 512:
+                        t5_embedding = torch.cat([t5_embedding, torch.zeros(512 - t5_embedding.shape[0], 1024)], dim=0)
+                    t5_embeddings.append(t5_embedding)
+                    caption_viewid = self.caption_view_idx_map[view_index]
+                    view_indices_conditioning.append(torch.ones(video.shape[1]) * caption_viewid)
+
+                    if self.ctrl_type:
+                        v_ctrl_data = self._load_control_data(
+                            {
+                                "ctrl_path": os.path.join(
+                                    self.dataset_dir,
+                                    view_key,
+                                    self.ctrl_data_pth_config["folder"],
+                                    f"{video_name}.{self.ctrl_data_pth_config['format']}",
+                                )
+                                if self.ctrl_data_pth_config["folder"] is not None
+                                else None,
+                                "frame_ids": frame_ids,
+                            }
+                        )
+                        if v_ctrl_data is None:  # Control data loading failed
+                            index = np.random.randint(len(self.video_paths))
+                            continue
```

We should raise an exception here. This continue statement is inside a view_index loop nested under the outer max_retries loop so it will have the impact of loading the next view from the same video_path, instead of moving on to the next video_path at the newly sampled index.
The catch block will reset index = np.random.randint(len(self.video_paths)) as well so we can also remove it from here.